### PR TITLE
Refactor PostController to address TA comments

### PR DESCRIPTION
**Pull request by yowang20**

### CODE REVIEW

File: backend/src/main/java/controllers/PostController.java [hunk 1]
```diff
@@ -69,23 +69,18 @@ protected void browsePost(Post selectedPost) {
...
         for (String ingredient: splitIngredients) {
             String strippedIngredient = ingredient.stripLeading();
-            if (!Character.isDigit(strippedIngredient.charAt(0))) {
-                inOut.setOutput("Invalid input. Did not specify numerical amount of one of the ingredients. Please enter countable ingredients again in correct format.");
-                countableInput = getCountableIngredients(promptCountable);
-            } else if (strippedIngredient.split(" ").length < 2) {
-                inOut.setOutput(errorMessage);
-                countableInput = getCountableIngredients(promptCountable);
-            } else if (strippedIngredient.contains("/")) {
+            if (checkInputInvalidity(strippedIngredient) || strippedIngredient.split(" ").length < 2) {
```
Comment (ericjest) [inline backend/src/main/java/controllers/PostController.java:83]: I think a better name would be "isInputInvalid" to reflect the boolean return value

File: backend/src/main/java/controllers/PostController.java [hunk 2]
```diff
@@ -256,7 +250,8 @@ private String getPostLikes(String id) {
      */
     private String getPostComments(String id) {
         String[] comments = postManager.getPostComments(id);
-        StringBuilder formattedComments = new StringBuilder("Comments:\n\n");
+        int numComments = comments.length;
+        StringBuilder formattedComments = new StringBuilder("Comments (").append(numComments).append("):\n\n");
```
Comment (ericjest) [inline backend/src/main/java/controllers/PostController.java:254]: could we just call comments.length on this line? Is there a reason we need to save numComments?

**Review by ericjest — CHANGES_REQUESTED** [review]

File: backend/src/main/java/controllers/PostController.java (hunk 1 above)
Comment (yowang20) [reply backend/src/main/java/controllers/PostController.java]: great point

**Review by shawnpl7 — APPROVED** [review]

**Comment by ericjest** [comment]
@yowang20 if you resolved the comments, feel free to merge

**Pull request merged**